File: Scala/code/src/main/scala/LogicalProblems/List/DigitsFromNumber.scala
package LogicalProblems.List

import scala.io.StdIn

/**
 * Write a function that takes a number and returns a list of its digits.
 * For Example:
 * Input 2342 should return [2,3,4,2].
 */
object DigitsFromNumber extends App {

  val infoMessage = s"Please enter a number greater tha 0"
  println(infoMessage)
  val input = StdIn.readInt()

  val output = getListOfDigits(input)
  println(output)

  private def getListOfDigits(num: Int) : List[Int] = {
    if(num > 0)
      getListOfDigits(num/10) :+ (num % 10)
    else
      Nil
  }

}

File: python/code/src/basicProblems/DigitsFromNumber.py
"""
    Accept a number as input and return a list of all the digits in the number.
    (Repeated digits in number should repeat in list as well)
"""

def getListOfDigits(num: int):
    if num is not None:
        ret_list = []
        while num > 0:
            last = num % 10
            ret_list.insert(0, last)
            num = num // 10

        return ret_list

if __name__ == '__main__':
    number = int(input("Enter a number: "))
    print(getListOfDigits(number))